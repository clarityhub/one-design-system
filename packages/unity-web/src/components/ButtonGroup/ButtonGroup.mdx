---
menu: Components
name: ButtonGroup
---

import { Playground, PropsTable } from 'docz';
import ButtonGroup from './ButtonGroup';

# ButtonGroup

A `ButtonGroup` groups a set of actions together that act like checkboxes or
radio inputs. For example, displaying days of the week would be better suited
as a ButtonGroup then as distinct checkboxes. 

```js
import ButtonGroup from '@clarityhub/unity-web/lib/components/ButtonGroup';
```

## Examples

<Playground>
    <ButtonGroup>
        <ButtonGroup.ButtonGroupItem>Mon</ButtonGroup.ButtonGroupItem>
        <ButtonGroup.ButtonGroupItem selected>Tue</ButtonGroup.ButtonGroupItem>
        <ButtonGroup.ButtonGroupItem>Wed</ButtonGroup.ButtonGroupItem>
        <ButtonGroup.ButtonGroupItem>Thu</ButtonGroup.ButtonGroupItem>
        <ButtonGroup.ButtonGroupItem selected>Fri</ButtonGroup.ButtonGroupItem>
        <ButtonGroup.ButtonGroupItem>Sat</ButtonGroup.ButtonGroupItem>
        <ButtonGroup.ButtonGroupItem>Sun</ButtonGroup.ButtonGroupItem>
    </ButtonGroup>
</Playground>

## Prop Types

The `\<ButtonGroup /\>` component takes the following prop types.

<PropsTable of={ButtonGroup} />

The `\<ButtonGroup.ButtonGroupItem /\>` component takes the following prop types.

<PropsTable of={ButtonGroup.ButtonGroupItem} />