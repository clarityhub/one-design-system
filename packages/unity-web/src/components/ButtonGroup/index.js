export { ButtonGroup, ButtonGroupItem } from './ButtonGroup';