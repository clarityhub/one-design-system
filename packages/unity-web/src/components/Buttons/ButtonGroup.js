import React from 'react';
import styled from '@emotion/styled';
import { bool } from 'prop-types';
import colors from '@clarityhub/unity-core/lib/colors';

import Button, { EmotionButton } from './Button';

const ButtonGroup = styled.div`
    box-shadow: ${colors.shadow.default};
    display: inline-block;

    ${EmotionButton} {
        box-shadow: none;
    }

    ${EmotionButton} + ${EmotionButton} {
        border-left-width: 0;
        border-left: 1px solid ${colors.muted.default};
    }
    ${EmotionButton}:not(:last-child):not(:first-of-type) {
        border-radius: 0;
    }
    ${EmotionButton}:first-of-type {
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
    }
    ${EmotionButton}:last-child {
        border-top-left-radius: 0;
        border-bottom-left-radius: 0;
    }
`;

const ButtonGroupItem = ({ selected, ...rest }) => (
	<Button type={selected ? 'primary' : 'default'} {...rest} />
);

ButtonGroupItem.propTypes = {
	selected: bool,
};

export { ButtonGroup, ButtonGroupItem };