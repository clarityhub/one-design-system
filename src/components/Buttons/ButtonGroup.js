import React from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import { bool } from 'prop-types';

import Button, { EmotionButton } from './Button';

const ButtonGroup = styled.div`
    ${EmotionButton} + ${EmotionButton} {
        border-left-width: 0;
    }
    ${EmotionButton}:not(:last-child):not(:first-of-type) {
        border-radius: 0;
    }
    ${EmotionButton}:first-of-type {
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
    }
    ${EmotionButton}:last-child {
        border-top-left-radius: 0;
        border-bottom-left-radius: 0;
    }
`;

const ButtonGroupItem = ({ selected, ...rest }) => (
    <Button type={selected ? 'primary' : 'default'} {...rest} />
);

ButtonGroupItem.propTypes = {
    selected: bool,
};

export { ButtonGroup, ButtonGroupItem };