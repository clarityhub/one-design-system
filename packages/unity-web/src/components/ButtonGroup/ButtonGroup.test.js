import React from 'react';
import renderer from 'react-test-renderer';

import ButtonGroup from './ButtonGroup';

describe('<ButtonGroup />', () => {
	it('renders a group of items', () => {
		const { ButtonGroupItem } = ButtonGroup;
		const node = (
			<ButtonGroup>
				<ButtonGroupItem>Mon</ButtonGroupItem>
				<ButtonGroupItem selected>Tue</ButtonGroupItem>
				<ButtonGroupItem>Wed</ButtonGroupItem>
				<ButtonGroupItem>Thu</ButtonGroupItem>
				<ButtonGroupItem selected>Fri</ButtonGroupItem>
				<ButtonGroupItem>Sat</ButtonGroupItem>
				<ButtonGroupItem>Sun</ButtonGroupItem>
			</ButtonGroup>
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});