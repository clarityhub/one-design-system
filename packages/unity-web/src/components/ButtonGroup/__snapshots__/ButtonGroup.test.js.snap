// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup /> renders a group of items 1`] = `
<div
  className="css-es8daz-ButtonGroup e1yduq3m0"
>
  <button
    className="-default css-1qfeqqs-Button-Button entsyi60"
    type="button"
  >
    Mon
  </button>
  <button
    className="-primary css-1qfeqqs-Button-Button entsyi60"
    type="button"
  >
    Tue
  </button>
  <button
    className="-default css-1qfeqqs-Button-Button entsyi60"
    type="button"
  >
    Wed
  </button>
  <button
    className="-default css-1qfeqqs-Button-Button entsyi60"
    type="button"
  >
    Thu
  </button>
  <button
    className="-primary css-1qfeqqs-Button-Button entsyi60"
    type="button"
  >
    Fri
  </button>
  <button
    className="-default css-1qfeqqs-Button-Button entsyi60"
    type="button"
  >
    Sat
  </button>
  <button
    className="-default css-1qfeqqs-Button-Button entsyi60"
    type="button"
  >
    Sun
  </button>
</div>
`;
